contrib/coredns/policy/policy.go

```diff
@@ -150,7 +174,47 @@ func (p *PolicyMiddleware) ServeDNS(ctx context.Context, w dns.ResponseWriter, r
 	}
 
 	if response.Permit {
-		return middleware.NextOrFailure(p.Name(), p.Next, ctx, w, r)
+		lr := new(NewLocalResponseWriter)
+		status,err := middleware.NextOrFailure(p.Name(), p.Next, ctx, lr, r)
+		if err != nil {
+			return status, err
+		}
+
+		state := request.Request{W: lr, Req: lr.Msg}
+
+		rr := lr.Msg.Answer[0]
+		switch state.Family() {
+		case 1:
+			attrs = append(attrs, &pb.Attribute{Id: "address", Type: "address", Value: rr.(*dns.A).A.String()})
+		case 2:
+			attrs = append(attrs, &pb.Attribute{Id: "address", Type: "address", Value: rr.(*dns.AAAA).AAAA.String()})
+
+		}
+
+		attrs[0].Value = "response"
+
+		var response Response
+		err = p.pdp.Validate(ctx, pb.Request{Attributes: attrs}, &response)
```

Comment:
We also will need to copy the `policy_id` attribute out of the original response obligations.